src/managers.py

```diff
@@ -0,0 +1,69 @@
+from git import Repo
+from joblib import Parallel, delayed
+from tqdm import tqdm
+
+from src.constants import n_jobs, VENDOR_PATH
+from src.models import File
+from src.providers import LanguagesProvider
+from src.services import GitService, GithubService, JsonService
+from src.setup import setup
+
+
+class DeveloperVectorManager:
+    """
+    A class to manage developer vectors.
+    """
+
+    def __init__(
+            self,
+            git_service: GitService,
+            github_service: GithubService,
+    ) -> None:
+        self.git_service = git_service
+        self.github_service = github_service
+
+    def clone_all_user_repos(self, username: str) -> [Repo]:
```

Comment:
the pipeline to get top N repositories should be like this:
* start with 1 repository (I think we discussed `sklearn`)
* get all stargazers for this repository - list of users
* for each get all stars - it will give us counter like `{ "repo_1": cnt_1, "repo_2": cnt_2, ...}`
* select top N repositories (probably 100 - adjust this number if it takes too long to process)